BASE/src/ServerTelemetryChannel/Implementation/Transmitter.cs

```diff
@@ -161,6 +162,34 @@ internal virtual void Enqueue(Transmission transmission)
 
             if (this.Buffer.Enqueue(transmissionGetter))
             {
+                if (transmission.HasFlushTask)
+                {
+                    lock (this.lockObj)
+                    {
+                        int temp_maxSenderCapacity = this.maxSenderCapacity;
+                        int temp_maxBufferCapacity = this.maxBufferCapacity;
+                        this.maxSenderCapacity = 0;
+                        this.maxBufferCapacity = 0;
+                        this.Storage.IsEnqueueSuccess = false;
+
+                        // Move buffer to storage
```

Comment:
if the intention is to move everything to storage, that can done directly, without calling ApplyPolicies and setting capacity to 0.